src/site/content/en/blog/codelab-address-form-best-practices/index.md

```diff
@@ -0,0 +1,285 @@
+
+{% Aside %}
+Ever wondered how to delete autofill suggestions in Chrome?
+* Windows: `Shift` + `delete`
+* Mac: `Shift` + `fn` + `delete`
+{% endAside %}
+
+You should notice a subtle change: with `autocomplete="name"`, the suggestions are now specific 
+values that were used previously in form inputs that also had `autocomplete="name"`. The browser 
+isn't just guessing what might be appropriate: you have control. You'll also see the **Manage…** 
+option to view and edit the names and addresses stored by your browser.
+
+<figure class="w-figure">
+  <img 
+    src="images/autofill-autocomplete.jpg" 
+    alt="Two screenshots of Chrome on an Android phone showing a form with a single input, with 
+    and without an autocomplete value. One shows browser UI heuristicically suggestions values; the 
+    other shows UI when there are stored autocomplete values.">
+  <figcaption class="w-figcaption">UI for autofill with guessed values, versus autocomplete.</figcaption>
+</figure>
+
+{% Aside %}
+If an [appropriate autocomplete value](/payment-and-address-form-best-practices/#autocomplete-attribute) 
+is available for an `input`, `select` or `textarea`, you should use it!
+{% endAside %}
+
+Now add [constraint validation attributes](https://developer.mozilla.org/en-US/docs/Web/Guide/HTML/HTML5/Constraint_validation) 
+`maxlength`, `pattern` and `required` so your input code looks like this:
+
+```html/2
+<section>
+  <label for="name">Name</label>
+  <input id="name" name="name" autocomplete="name" 
+    maxlength="100" pattern="[\p{L} \-\.]+" required>
+</section>
+```
+
+* `maxlength="100"` means the browser won't allow any input longer than 100 characters.
+
+`pattern="[\p{L} \-\.]+"` uses a regular expression that allows [Unicode letter characters](https://javascript.info/regexp-unicode), 
+hyphens and periods (full stops). That means names such as Françoise or Jörg aren't classed as 
+'invalid'. That isn't the case if you use the value `\w` which [only allows characters from the 
+[Latin alphabet](/payment-and-address-form-best-practices/#unicode-matching).
+
+`required` means… required! The browser will not allow the form to be submitted without data for 
+this field, and will warn and highlight the input if you attempt to submit it. No extra code 
+required!
+
+{% Aside 'caution' %}
+This codelab doesn't address 😜 localization or internationalization. Depending on where your users 
+are located, you need to consider address **formats** as well as the different **names** used for 
+address components, even within the same language: ZIP, postal code, Eircode or PIN? It may be 
+necessary for your site to [customize for multiple locales](https://www.smashingmagazine.com/2020/11/internationalization-localization-static-sites#determining-user-s-language-and-region), 
+but the address form in this codelab is designed for flexibility, and should work 'well enough' for 
+a range of addresses.
```

Comment:
```suggestion
This codelab doesn't address (😜) localization or internationalization. Depending on where your users 
are located, you need to consider address *formats* as well as the different *names* used for 
address components, even within the same language: ZIP, postal code, Eircode or PIN? It may be 
necessary for your site to [customize for multiple locales](https://www.smashingmagazine.com/2020/11/internationalization-localization-static-sites#determining-user-s-language-and-region), 
but the address form in this codelab is designed for flexibility, and should work "well enough" for 
a range of addresses.
```